client/rest-high-level/src/main/java/org/elasticsearch/client/security/DelegatePkiAuthenticationRequest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.Validatable;
+import org.elasticsearch.client.ValidationException;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Collections.unmodifiableList;
+
+public final class DelegatePkiAuthenticationRequest implements Validatable, ToXContentObject {
+
+    private final List<X509Certificate> x509CertificateChain;
+
+    public DelegatePkiAuthenticationRequest(final List<X509Certificate> x509CertificateChain) {
+        if (x509CertificateChain == null || x509CertificateChain.isEmpty()) {
+            throw new IllegalArgumentException("certificate chain must not be empty or null");
+        }
+        this.x509CertificateChain = unmodifiableList(x509CertificateChain);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        final List<String> encodedCertificates = new ArrayList<>(x509CertificateChain.size());
+        try {
+            for (int i = 0; i < x509CertificateChain.size(); i++) {
+                encodedCertificates.add(Base64.getEncoder().encodeToString(x509CertificateChain.get(i).getEncoded()));
+            }
+        } catch (CertificateEncodingException e) {
+            throw new IOException(e);
+        }
+        builder.startObject()
+            .field("x509_certificate_chain", encodedCertificates);
```

Comment:
Nice! Thank you for the suggestion.
I prefer the explicit option (`builder.value(Base64.getEncoder().encodeToString(cert.getEncoded()));`), even though it is more verbose, because it also serves as documentation.